core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java

```diff
@@ -72,11 +61,11 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         currentFieldName = parser.currentName();
                     } else {
                         if ("value".equals(currentFieldName)) {
-                            value = parser.objectBytes();
+                            value = parser.text();
```

Comment:
before the refactoring the parser holds most of the code. builders are used only from the java api and what they do is they allow you to set stuff which gets then serialized in json format anyway for the parser to parse it on each data node. if the parser previously supported objects, but the builder only strings, that is a bug in the java api to me. What rules is the parser, and we should do what the parser supported in our new builder, which will be also used for communication between the nodes when the refactoring is done, meaning that if we keep string the bug will not only be in the java api but in es core in general.